tests/src/test/java/alluxio/client/cli/fs/command/LoadMetadataCommandV2IntegrationTest.java

```diff
@@ -0,0 +1,295 @@
+    FileSystemShell anotherFsShell = new FileSystemShell(Configuration.global());
+
+    // Cancel a running task
+    anotherFsShell.run("loadMetadata", "-v2", "-o", "cancel", "-id", id);
+    assertTrue(mOutput.toString().contains(String.format("Task group %s cancelled", id)));
+
+    // Cancel a canceled task
+    mOutput.reset();
+    anotherFsShell.run("loadMetadata", "-v2", "-o", "cancel", "-id", id);
+    assertTrue(mOutput.toString().contains(String.format("Task %s not found or has already been canceled", id)));
+
+    // Trying to cancel completed task
+    mOutput.reset();
+    mFsShell.run("loadMetadata", "-v2", "-R", "-a", uriDir.toString());
+    matcher = pattern.matcher(mOutput.toString());
+    while(!matcher.find()) {
+      matcher = pattern.matcher(mOutput.toString());
+    }
+    id = matcher.group(1);
+    mFsShell.run("loadMetadata", "-v2", "-o", "cancel", "-id", id);
+    assertTrue(mOutput.toString().contains(String.format("Task %s not found or has already been canceled", id)));
+  }
+
+  @Test
+  public void loadMetadataTestV2R() throws IOException, AlluxioException {
+    // dir number
+    int dirCount = 3;
+    // Child number for one dir
+    int fileCount = 10;
+    for (int dirIndex = 0; dirIndex < dirCount; dirIndex++) {
+      for (int fileIndex = 0; fileIndex < fileCount; fileIndex++) {
+        mS3Client.putObject(TEST_BUCKET, "test" + dirIndex + "/" + fileIndex, TEST_CONTENT);
+      }
+    }
+    AlluxioURI uriDir = new AlluxioURI("/" );
+
+    mFsShell.run("loadMetadata", "-v2", "-a", uriDir.toString());
+    assertTrue(mOutput.toString().contains("State: SUCCEEDED"));
+    GetStatusPOptions getStatusPOptions =
+        GetStatusPOptions.newBuilder().setLoadMetadataType(LoadMetadataPType.NEVER).build();
+    URIStatus statusAfter = null;
+    for (int i = 0; i < dirCount; i++) {
+      statusAfter = mFileSystem.getStatus(new AlluxioURI("/test" + i), getStatusPOptions);
+      assertEquals(statusAfter.getFileInfo().getLength(), 0);
+    }
+    // avoid legacy "State: SUCCEEDED"
+    mOutput.reset();
+    mFsShell.run("loadMetadata", "-v2", "-R", "-a", uriDir.toString());
+    assertTrue(mOutput.toString().contains("State: SUCCEEDED"));
+    for (int i = 0; i < dirCount; i++) {
+      statusAfter = mFileSystem.getStatus(new AlluxioURI("/test" + i), getStatusPOptions);
+      assertEquals(statusAfter.getFileInfo().getLength(), fileCount);
+    }
+  }
+
+  @Test
+  public void loadMetadataTestV2HeavyLoad() throws IOException, AlluxioException {
+    int fileCount = 1000000;
+    for (int i = 0; i < fileCount; i++) {
+      mS3Client.putObject(TEST_BUCKET, TEST_FILE + i, TEST_CONTENT);
+    }
+    mOutput.reset();
+    AlluxioURI uriDir = new AlluxioURI("/" );
+    mFsShell.run("loadMetadata", "-v2", "-R", "-a", uriDir.toString());
+    assertTrue(mOutput.toString().contains("State: SUCCEEDED"));
+    assertTrue(mOutput.toString().contains(String.format("Success op count={[CREATE:%d]}", fileCount)));
```

Comment:
remove this
do a fsMaster.listStatus() instead

Reply:
is FileSystem.getStatus() ok?